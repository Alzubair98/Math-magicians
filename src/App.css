body {
  padding: 15% 40% 0 40%;
}

.calculator_container {
  border-style: solid;
  border-width: 1px;
  width: 20rem;
  height: 18rem;
  display: flex;
  flex-direction: column;
}

.result_div {
  display: flex;
  justify-content: end;
  align-items: flex-end;
  font-size: 2rem;
  height: 3rem;
  background-color: gray;
  color: white;
}

.first_row {
  display: flex;
  height: 3rem;
}

.button {
  display: flex;
  justify-content: center;
  align-items: center;
  border-style: solid;
  border-width: 0.5px;
  border-color: aliceblue;
  width: 25%;
  background-color: antiquewhite;
}

.operation_button {
  background-color: orange;
}

.zero_button {
  width: 50%;
}
