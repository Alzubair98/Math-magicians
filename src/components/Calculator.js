import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [obj, addNewObject] = useState({ total: null, next: null, operation: null });

  const result = (event) => {
    const newObject = calculate(obj, event.target.textContent);
    addNewObject(newObject);
  };

  return (
    <div className="calculator_container">
      <div className="result_div">{obj.next || obj.total || 0}</div>
      <div className="first_row">
        <button type="button" className="button" onClick={result}>AC</button>
        <button type="button" className="button" onClick={result}>+/-</button>
        <button type="button" className="button" onClick={result}>%</button>
        <button type="button" className="button operation_button" onClick={result}>+</button>
      </div>
      <div className="first_row">
        <button type="button" className="button" onClick={result}>7</button>
        <button type="button" className="button" onClick={result}>8</button>
        <button type="button" className="button" onClick={result}>9</button>
        <button type="button" className="button operation_button" onClick={result}>x</button>
      </div>
      <div className="first_row">
        <button type="button" className="button" onClick={result}>4</button>
        <button type="button" className="button" onClick={result}>5</button>
        <button type="button" className="button" onClick={result}>6</button>
        <button type="button" className="button operation_button" onClick={result}>-</button>
      </div>
      <div className="first_row">
        <button type="button" className="button" onClick={result}>1</button>
        <button type="button" className="button" onClick={result}>2</button>
        <button type="button" className="button" onClick={result}>3</button>
        <button type="button" className="button operation_button" onClick={result}>+</button>
      </div>
      <div className="first_row">
        <button type="button" className="button zero_button" onClick={result}>0</button>
        <button type="button" className="button" onClick={result}>.</button>
        <button type="button" className="button operation_button" onClick={result}>=</button>
      </div>
    </div>

  );
};

export default Calculator;
