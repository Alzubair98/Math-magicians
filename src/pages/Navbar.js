import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 1,
      path: '/Calculator',
      text: 'Calculator',
    },
    {
      id: 1,
      path: '/Quote',
      text: 'Quote',
    },
  ];

  return (
    <div>
      {links.map((link) => (
        <li className="navbar-item" key={link.id}>
          <NavLink to={link.path} className="active-link">{link.text}</NavLink>
        </li>
      ))}
      <h1>Math Magicians</h1>
    </div>
  );
};

export default Navbar;
