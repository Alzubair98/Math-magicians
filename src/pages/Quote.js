import React from 'react';

const Quote = () => (
  <div className="Quote_continer">
    <h1>
      “Mathematics is the most beautiful and most powerful
      creation of the human spirit.” – Stefan Banach
    </h1>
  </div>
);
export default Quote;
