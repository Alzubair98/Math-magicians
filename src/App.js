import React from 'react';
import './App.css';
import { Routes, Route } from 'react-router-dom';
import Calculator from './components/Calculator';
import Navbar from './pages/Navbar';
import Home from './pages/home';
import Quote from './pages/Quote';

const App = () => (
  <>
    <Navbar />
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/Calculator" element={<Calculator />} />
      <Route path="/Quote" element={<Quote />} />
    </Routes>
  </>
);

Calculator.defalutValue = {};
export default App;
