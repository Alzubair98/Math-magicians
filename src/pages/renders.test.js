import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from './home';
import Quote from './Quote';
import Navbar from './Navbar';
import Calculator from '../components/Calculator';
import App from '../App';

describe('Navbar , home & Quote renderer test', () => {
  it('matches the snapshot for home', () => {
    const tree = renderer.create(<Home />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('matches the snapshot for Quote', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches the snapshot for caculator', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  
});
