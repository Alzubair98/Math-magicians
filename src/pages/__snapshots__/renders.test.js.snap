// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar , home & Quote renderer test matches the snapshot for Navbar 1`] = `
<div
  className="nav-continer"
>
  <h1
    className="math-magicians"
  >
    Math Magicians
  </h1>
  <div
    className="options-continer"
  >
    <li
      className="navbar-item"
    >
      <a
        aria-current="page"
        className="active-link active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="active-link"
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="active-link"
        href="/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </div>
</div>
`;

exports[`Navbar , home & Quote renderer test matches the snapshot for Navbar 2`] = `
Array [
  <div
    className="nav-continer"
  >
    <h1
      className="math-magicians"
    >
      Math Magicians
    </h1>
    <div
      className="options-continer"
    >
      <li
        className="navbar-item"
      >
        <a
          aria-current="page"
          className="active-link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          className="active-link"
          href="/Calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          className="active-link"
          href="/Quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </div>
  </div>,
  <div
    className="Home-continare"
  >
    <h3
      className="welcome_text"
    >
      Welcome to our page!
    </h3>
    <p
      className="paragraph"
    >
      Mathematics from Ancient Greek μάθημα; máthēma: knowledge, study, learning is an area of knowledge that includes such topics as numbers arithmetic, number theory, formulas and related structures algebra, shapes and the spaces in which they are contained geometry, and quantities and their changes calculus and analysis. Most mathematical activity involves the use of pure reason to discover or prove the properties of abstract objects, which consist of either abstractions from nature or—in modern mathematics—entities that are stipulated with certain properties, called axioms. A mathematical proof consists of a succession of applications of some deductive rules to already known results, including previously proved theorems, axioms and in case of abstraction from nature some basic properties that are considered as true starting points of the theory under consideration
    </p>
    <p
      className="paragraph"
    >
      Mathematics is used in science for modeling phenomena, which then allows predictions to be made from experimental laws. The independence of mathematical truth from any experimentation implies that the accuracy of such predictions depends only on the adequacy of the model. Inaccurate predictions, rather than being caused by incorrect mathematics, imply the need to change the mathematical model used. For example, the perihelion precession of Mercury could only be explained after the emergence of Einstein's general relativity, which replaced Newton's law of gravitation as a better mathematical model.
    </p>
  </div>,
]
`;

exports[`Navbar , home & Quote renderer test matches the snapshot for Quote 1`] = `
<div
  className="Quote_continer"
>
  <h1>
    “Mathematics is the most beautiful and most powerful creation of the human spirit.” – Stefan Banach
  </h1>
</div>
`;

exports[`Navbar , home & Quote renderer test matches the snapshot for caculator 1`] = `
<div
  className="container"
>
  <div
    className="text-container"
  >
    <h2>
      Lets Do Some Math!
    </h2>
  </div>
  <div
    className="calculator_container"
  >
    <div
      className="result_div"
      data-testid="result"
    >
      0
    </div>
    <div
      className="first_row"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button operation_button"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
    </div>
    <div
      className="first_row"
    >
      <button
        className="button"
        data-testid="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button operation_button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="first_row"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button operation_button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="first_row"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button operation_button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="first_row"
    >
      <button
        className="button zero_button"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button operation_button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Navbar , home & Quote renderer test matches the snapshot for home 1`] = `
<div
  className="Home-continare"
>
  <h3
    className="welcome_text"
  >
    Welcome to our page!
  </h3>
  <p
    className="paragraph"
  >
    Mathematics from Ancient Greek μάθημα; máthēma: knowledge, study, learning is an area of knowledge that includes such topics as numbers arithmetic, number theory, formulas and related structures algebra, shapes and the spaces in which they are contained geometry, and quantities and their changes calculus and analysis. Most mathematical activity involves the use of pure reason to discover or prove the properties of abstract objects, which consist of either abstractions from nature or—in modern mathematics—entities that are stipulated with certain properties, called axioms. A mathematical proof consists of a succession of applications of some deductive rules to already known results, including previously proved theorems, axioms and in case of abstraction from nature some basic properties that are considered as true starting points of the theory under consideration
  </p>
  <p
    className="paragraph"
  >
    Mathematics is used in science for modeling phenomena, which then allows predictions to be made from experimental laws. The independence of mathematical truth from any experimentation implies that the accuracy of such predictions depends only on the adequacy of the model. Inaccurate predictions, rather than being caused by incorrect mathematics, imply the need to change the mathematical model used. For example, the perihelion precession of Mercury could only be explained after the emergence of Einstein's general relativity, which replaced Newton's law of gravitation as a better mathematical model.
  </p>
</div>
`;
